src/main/java/sparksoniq/spark/iterator/flowr/ForClauseSparkIterator.java

```diff
@@ -209,56 +209,51 @@ public void close() {
             JavaRDD<Row> rowRDD = initialRdd.map(new ForClauseSerializeClosure());
 
             // apply the schema to row RDD
-            this._df = SparkSessionManager.getInstance().getOrCreateSession().createDataFrame(rowRDD, schema);
+            return SparkSessionManager.getInstance().getOrCreateSession().createDataFrame(rowRDD, schema);
+        }
 
-        } else {        //if it's not a start clause
-            if (_child.isDataFrame()) {
-                this._df = this._child.getDataFrame(context);
+        if (_child.isDataFrame()) {
+            Dataset<Row> df = this._child.getDataFrame(context);
 
-                // Add new column name to existing schema
-                String[] columnNamesPrimitive = this._df.schema().fieldNames();
-                ArrayList<String> columnNames = new ArrayList<>(Arrays.asList(columnNamesPrimitive));
-                columnNames.add(_variableName);
+            StructType oldSchema = df.schema();
+            StructType newSchema = oldSchema.add(_variableName, DataTypes.BinaryType);
```

Comment:
Does this code also work with variable hiding?

Test:
for $i in parallelize((1 to 3)) for $i in 4 to 6 return $i
which returns 4 5 6 4 5 6 4 5 6

If the variable is hidden, you may need to replace, rather than add the variable in the schema.

Reply:
Both the alpha and the dataframes (to my surprise) show the identical behavior given below. I will be adding test cases for variable hiding as well.
```
jiqs$ for $i in parallelize((1 to 3)) for $i in 4 to 6 return $i
>>>
>>>
(4, 5, 6, 4, 5, 6, 4, 5, 6)
jiqs$ for $i in parallelize((1 to 5)) for $i in 4 to 6 return $i
>>>
>>>
(4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6)
jiqs$ for $i in parallelize((1 to 5)) for $i in 4 to 5 return $i
>>>
>>>
(4, 5, 4, 5, 4, 5, 4, 5, 4, 5)
```

I believe this works out of the box due to our usage of Dynamic context as shown below:
```
// Taken from ForClauseMapClosure.java Line 55
// Create dynamic context with deserialized data
DynamicContext context = new DynamicContext();
for (int columnIndex = 0; columnIndex < columnNames.length; columnIndex++) {
    context.addVariableValue(columnNames[columnIndex], rowColumns.get(columnIndex));
}
```
The second column with the identical name overwrites the existing value in the Dynamic context's hashmap (addVariableValue calls put function of Hashmap -> Java Doc: https://docs.oracle.com/javase/8/docs/api/java/util/Hashtable.html#put-K-V-)

Should we replace the variable or is the current implementation good?